### DISCUSSION TITLE
Building Sphinx documentation offline, prefetch remote images

### DISCUSSION THREAD

**Discussion by Thaodan** [sphinx-doc/sphinx#12721] [Q&A]
Hey,

I'm trying to build sphinx documentation offline in a OBS-worker (open build service).
Is it possible to somehow prefetch remote images so Sphinx doesn't fail embedding the images when building offline?

**Comment by AA-Turner** [comment]
The build shouldn't fail offline, rather it ought merely emit a warning.

Perhaps you could hook into `requests.get`? There's no way to prefetch though, as we don't know what the images are until we do the build. (You could try a regular expression on image directives, perhaps?)

A

**Reply by Thaodan** [reply]
Could the syntax break if the images are not present? I'm building Info pages, the texinfo outputted by Sphinx was wrong in my case.

I looked at the Sphinx code, it looks if an image already has downloaded, if it was possible to have alternative locations of images used in the documents that would work.
